### PR TITLE
feat(usecases): don't create use cases anymore on the fly (#2745)

### PR DESCRIPTION
**Pull request by carinelebas**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.
>
> ℹ️ **Feature environment** — A dedicated preview environment (`https://dev-pr-{number}.hub.staging.filigran.io`) is automatically deployed when CI passes. Add the `skip-feature-env` label to opt out.

Don't create anymore use cases when ingesting connectors if they are not already created.
Seed use cases for test to avoid creating them each time.

Ingest a connector V1 and V2 with unknown use cases. Check the use case is not created.

## Related issues

- Related to #2745

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code
- [ ] I added the `skip-feature-env` label if a feature environment is not needed for this PR

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/backend/tests/tests.const.ts [hunk 1]
```diff
@@ -144,6 +145,32 @@ export const SERVICES = {
   },
 };
 
+// Reusable use cases seeded in tests/seeds/04-insert-use-cases.js, to avoid
+// ad hoc creation in every test file. Names are not significant to the
+// features under test.
+export const TEST_USE_CASES = {
```
Comment (Kroustille) [inline apps/backend/tests/tests.const.ts:151]: 💯

File: apps/backend/tests/seeds/04-insert-use-cases.js [hunk 2]
```diff
@@ -7,6 +7,36 @@ export async function seed(knex) {
         color: '#f5a623',
         product: ['opencti', 'openaev'],
       },
+      {
+        id: '9d3a9c3d-3b9a-4b5b-8b0a-1a2b3c4d5e01',
+        name: 'Automation',
+        color: '#0099cc',
+        product: ['opencti', 'openaev'],
```
Comment (Kroustille) [inline apps/backend/tests/seeds/04-insert-use-cases.js]: Would it be useful to have use cases with only one product ? 😄

File: apps/backend/src/modules/use-case/use-case.domain.test.ts [hunk 3]
```diff
@@ -104,4 +104,36 @@ describe('useCaseDomain', () => {
...
+
+      const found =
+        await useCaseDomain.loadUseCaseByNameCaseInsensitive(
+          'existing usecase'
+        );
+
+      expect(found?.id).toBe(created.id);
+    });
+
+    it('should not treat % and _ in the provided name as SQL wildcards', async () => {
```
Comment (Kroustille) [inline apps/backend/src/modules/use-case/use-case.domain.test.ts]: The test mentions % and _ but only tests _

File: apps/backend/src/modules/use-case/use-case.app.ts [hunk 4]
```diff
@@ -36,15 +37,24 @@ export const useCaseApp = {
...
+      }
       insertObjectUseCase.push({
         object_id: objectId,
         use_case_id: useCase.id,
       });
     }
+
+    if (!insertObjectUseCase.length) {
+      return;
+    }
```
Comment (Kroustille) [inline apps/backend/src/modules/use-case/use-case.app.ts]: We may want to have this check in `objectUseCaseDomain.insertObjectUseCase` to make sure this case never happens 🤔

File: apps/backend/src/modules/use-case/use-case.domain.test.ts [hunk 5]
```diff
@@ -104,4 +104,36 @@ describe('useCaseDomain', () => {
...
+      await TestHelper.useCase.create({
+        name: 'AutomationX',
+        color: '#aaaaaa',
+      });
+
+      const found =
+        await useCaseDomain.loadUseCaseByNameCaseInsensitive('Automation_');
+
+      expect(found).toBeUndefined();
```
Comment (hervyt) [inline apps/backend/src/modules/use-case/use-case.domain.test.ts]: **nitpick**: You can parametrize these tests

File: apps/backend/src/modules/use-case/use-case.app.ts
Comment (Kroustille) [inline apps/backend/src/modules/use-case/use-case.app.ts]: This function is only used in tests and can be removed.

File: apps/backend/src/modules/shareable-resource/opencti/integration/ingest-manifest/ingest-manifest.domain.test.ts [hunk 6]
```diff
@@ -118,8 +122,8 @@ describe('upsertConnectors', () => {
         const useCaseNames = useCases.map((useCase) => useCase.name);
 
         expect(useCaseNames).toHaveLength(2);
-        expect(useCaseNames).toContain('automation');
-        expect(useCaseNames).toContain('integration');
+        expect(useCaseNames).toContain(TEST_USE_CASES.AUTOMATION.NAME);
+        expect(useCaseNames).toContain(TEST_USE_CASES.INTEGRATION.NAME);
```
Comment (hervyt) [inline apps/backend/src/modules/shareable-resource/opencti/integration/ingest-manifest/ingest-manifest.domain.test.ts:126]: 💯

File: apps/backend/src/modules/use-case/use-case.app.test.ts [hunk 7]
```diff
@@ -124,11 +125,18 @@ describe('use Case app', () => {
...
     beforeEach(async () => {
       // Object_UseCase rows must be cleared first: UseCase deletion is
       // blocked by the foreign key while links still reference it.
       await TestHelper.objectUseCase.delete({});
       await TestHelper.useCase.delete({});
+      warnSpy = vi.spyOn(logApp, 'warn').mockImplementation(() => {});
+    });
+
+    afterEach(() => {
+      warnSpy.mockRestore();
```
Comment (Kroustille) [inline apps/backend/src/modules/use-case/use-case.app.test.ts]: This restore is not required because we already `restoreAllMocks` in `tests/setup-test.ts` afterEach

**Review by Kroustille — COMMENTED** [review]
This is really an edge case but if there is twice the same use case but with different cases (`exemple` and `Exemple`) in a connector manifest, the entire document creation will fail because it resolves to the same use case id and will break the object use case primary key.

**Review by hervyt — APPROVED** [review]

**Review by Kroustille — APPROVED** [review]

File: apps/backend/src/modules/use-case/use-case.domain.test.ts (hunk 5 above)
Comment (carinelebas) [reply apps/backend/src/modules/use-case/use-case.domain.test.ts]: done in [2d3424f](https://github.com/XTM-Hub/xtm-hub/pull/3052/commits/2d3424faf3d122033d704d56e5c40fd921336f2c)

File: apps/backend/src/modules/use-case/use-case.app.test.ts (hunk 7 above)
Comment (carinelebas) [reply apps/backend/src/modules/use-case/use-case.app.test.ts]: removed in [2d3424f](https://github.com/XTM-Hub/xtm-hub/pull/3052/commits/2d3424faf3d122033d704d56e5c40fd921336f2c)

File: apps/backend/src/modules/use-case/use-case.app.ts
Comment (carinelebas) [reply apps/backend/src/modules/use-case/use-case.app.ts]: removed in [2d3424f](https://github.com/XTM-Hub/xtm-hub/pull/3052/commits/2d3424faf3d122033d704d56e5c40fd921336f2c)

File: apps/backend/src/modules/use-case/use-case.app.ts (hunk 4 above)
Comment (carinelebas) [reply apps/backend/src/modules/use-case/use-case.app.ts]: done in [2d3424f](https://github.com/XTM-Hub/xtm-hub/pull/3052/commits/2d3424faf3d122033d704d56e5c40fd921336f2c)

File: apps/backend/src/modules/use-case/use-case.domain.test.ts (hunk 3 above)
Comment (carinelebas) [reply apps/backend/src/modules/use-case/use-case.domain.test.ts]: done in [2d3424f](https://github.com/XTM-Hub/xtm-hub/pull/3052/commits/2d3424faf3d122033d704d56e5c40fd921336f2c)

File: apps/backend/tests/seeds/04-insert-use-cases.js (hunk 2 above)
Comment (carinelebas) [reply apps/backend/tests/seeds/04-insert-use-cases.js]: not for now, but in case it's needed later, it's done in [2d3424f](https://github.com/XTM-Hub/xtm-hub/pull/3052/commits/2d3424faf3d122033d704d56e5c40fd921336f2c)
